src/components/Select/SelectBase.tsx

```diff
@@ -0,0 +1,273 @@
+import * as React from 'react';
+import Downshift, { GetItemPropsOptions, GetToggleButtonPropsOptions } from 'downshift';
+import bem from '../../utils/bem';
+import FieldWrapper from '../FieldWrapper';
+import { List } from '../List';
+import { ENTER_KEY } from '../../constants';
+import { Props as ListRendererProps } from './SuggestionsList';
+import styles from './SelectBase.scss';
+
+export interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** an array of objects that will be used to render the suggestions list. */
+    suggestions: object[];
+    /** suggestionToString(suggestion) should return a string to be displayed in the UI. e.g.: suggestion => suggestion.name */
+    suggestionToString: (suggestions?: object | null) => string;
+    /** trigger of the initial focus of the input field */
+    isInitiallyFocused?: boolean;
```

Comment:
The idea of setting focus by a prop can lead to unexpected behavior. There can be multiple `Select` components on a page, and if each of them has `isInitiallyFocused` set to `true`, which one would get focus in the end?

Reply:
good point. I copied this prop from Autofocus, but this can be achieved in the application using refs and so on. I will delete it.